xbmc/filesystem/PluginDirectory.cpp

```diff
@@ -486,35 +485,26 @@ bool CPluginDirectory::WaitOnScriptResult(const std::string &scriptPath, int scr
       CScriptObserver scriptObs(scriptId, m_fetchComplete);
       if (!CGUIDialogBusy::WaitOnEvent(m_fetchComplete, 200))
       {
-        cancelled = true;
+        m_cancelled = true;
       }
       scriptObs.Abort();
     }
   }
   else
   {
-    // kill the script if it does not return within 30 seconds
-    // and we are not waiting for user input
+    // Wait for directory fetch to complete, end, or be cancelled
+    while (!m_cancelled
```

Comment:
it will be attempted stopped when kodi shuts down, then this will terminate. I don't see what else can be done about it